course_catalog/serializers.py

```diff
@@ -504,42 +506,69 @@ def create(self, validated_data):
+        # 2) move the item into its new position
+        # this operation gets slower the further the item is moved, but it is sufficient for now
         with transaction.atomic():
-            instance.position = validated_data["position"]
+            if position > instance.position:
+                # move items between the old and new positions up, inclusive of the new position
+                UserListItem.objects.filter(
+                    position__lte=position, position__gt=instance.position
+                ).update(position=F("position") - 1)
+            else:
+                # move items between the old and new positions down, inclusive of the new position
+                UserListItem.objects.filter(
+                    position__lt=instance.position, position__gte=position
+                ).update(position=F("position") + 1)
+            # now move the item into place
+            instance.position = position
             instance.save()
+
         self.update_index(instance.user_list)
         return instance
 
     class Meta:
         model = UserListItem
         extra_kwargs = {"position": {"required": False}}
         exclude = ("created_on", "updated_on")
+        list_serializer_class = UserListItemListSerializer
 
 
-class SimpleUserListSerializer(
+class UserListSerializer(
     serializers.ModelSerializer, FavoriteSerializerMixin, ListsSerializerMixin
 ):
     """
     Simplified serializer for UserList model.
-    Uses the SimpleUserListItemSerializer for items, which contains only essential attributes.
     """
 
-    items = serializers.SerializerMethodField()
+    item_count = serializers.SerializerMethodField()
     topics = WriteableSerializerMethodField()
     author_name = serializers.SerializerMethodField()
     object_type = serializers.CharField(read_only=True, source="list_type")
-
-    def get_items(self, instance):
-        """get items"""
-        return [
-            SimpleUserListItemSerializer(item).data
-            for item in instance.items.select_related("content_type")
-        ]
+    image_src = serializers.SerializerMethodField()
 
     def get_author_name(self, instance):
         """get author name for userlist"""
         return instance.author.profile.name
 
+    def get_item_count(self, instance):
+        """Return the number of items in the list"""
+        return getattr(instance, "item_count", None) or instance.items.count()
+
+    def get_image_src(self, instance):
+        """Return the user list's image or the image of the first item"""
+        if instance.image_src:
+            return instance.image_src.url
+
+        # image_src is either a URLField or a ImageField
+        list_item = instance.items.order_by("position").first()
+        if list_item:
+            image_src = list_item.item.image_src
+            return image_src if isinstance(image_src, str) else image_src.url
```

Comment:
Might be just bad data on my instance, but this line raises an exception for one of my users:

```
web_1      |   File "./course_catalog/serializers.py", line 569, in get_image_src
web_1      |     return image_src if isinstance(image_src, str) else image_src.url
web_1      | AttributeError: 'NoneType' object has no attribute 'url'
```